src/zarr/store/local.py

```diff
@@ -68,7 +69,13 @@ def _put(
             f.write(value.as_numpy_array().tobytes())
         return None
     else:
-        return path.write_bytes(value.as_numpy_array().tobytes())
+        view = memoryview(value.as_numpy_array().tobytes())
```

Comment:
`pathlib.Path.write_bytes` doesn't provide control over the `mode`. So this is just that method inlined, with `mode="xb"` if we want exclusive create.